### PR TITLE
Paramfetch with aria2 support

### PR DESCRIPTION
**Pull request by magik6k**
This is a copy of https://github.com/filecoin-project/go-paramfetch/blob/master/paramfetch.go with:
* Heavily opinionated support for aria2
* Parallel fetch support

In the future we can think of adding support for p2p fetches from the cluster, but for now this makes my paramfetch operations 7-10x faster

### CODE REVIEW

**Review by LexLuthr — COMMENTED** [review]
We should update our dependencies to include aria2c here on Linux and MacOS systems.

**Comment by magik6k** [comment]
Yes, but not that it is technically optional in this PR, so nothing breaks if you don't have it, just the initial paramfetch is not as fast

**Comment by snadrus** [comment]
Please add:
  , aria2
to 
   File: apt/DEBIAN/control 
   Line: starts with "Depends:"

I've verified that name in the past 4 Ubuntu versions in the Universe repo.

**Review by LexLuthr — APPROVED** [review]
Tested. Looks good.

File: lib/fastparamfetch/paramfetch.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+	return err
+}
+
+func fetchWithAria2c(ctx context.Context, out, url string) error {
+	aria2cPath, err := exec.LookPath("aria2c")
+	if err != nil {
+		return xerrors.New("aria2c not found in PATH")
+	}
+
+	cmd := exec.CommandContext(ctx, aria2cPath, "--continue", "-x5", "-o", out, url)
```
Comment (LexLuthr) [inline lib/fastparamfetch/paramfetch.go:310]:
```suggestion
	cmd := exec.CommandContext(ctx, aria2cPath, "--continue", "-x16", "-s16", "-o", out, url)
```

**Review by LexLuthr — APPROVED** [review]
Just 1 one small change to speed up download

**Pull request merged**